components/admin/requests/processing/TaskCardUploadStep.tsx

```diff
@@ -0,0 +1,132 @@
+import { SyntheticEvent, useRef, useState, FC } from 'react';
+import { WarningTwoIcon, CloseIcon } from '@chakra-ui/icons';
+import { Button, Text, Link, HStack, VStack, IconButton, Box, Tooltip } from '@chakra-ui/react';
+
+// 5 MB file size limit
+const FILE_SIZE_LIMIT = 5 * 1024 * 1024;
+
+// File upload button props
+type Props = {
+  readonly fileUrl: string | null; // currently uploaded file url
+  readonly onUploadFile: (selectedFile: File) => void; // handle file upload
+  readonly isDisabled: boolean;
+  readonly onUndo: () => void;
+};
+
+/**
+ * POA form upload component allowing users to upload POA form PDF file
+ */
+export const TaskCardUploadStep: FC<Props> = ({ isDisabled, fileUrl, onUploadFile, onUndo }) => {
+  const hiddenFileInput = useRef<HTMLInputElement>(null);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  /**
+   * Handle upload button click
+   */
+  const handleClick = () => {
+    if (hiddenFileInput.current) {
+      hiddenFileInput.current.click();
+    }
+  };
+
+  /**
+   * Get file name from s3 URL
+   * Decode URL in case of file names with special characters
+   */
+  const getFileName = () => {
+    return fileUrl ? new URL(decodeURI(fileUrl)).pathname.split('/').at(-1) : null;
+  };
```

Comment:
We actually don't need to wrap this in a function! Anything in the function body is run on every render, and since we are calling this function on every render of this component, we can just pull out the logic.
```suggestion
  /** File name retrieved from s3 URL */
  const fileName = fileUrl ? new URL(decodeURI(fileUrl)).pathname.split('/').at(-1) : null;
```